src/neo-vm/ExecutionEngine.cs

```diff
@@ -852,8 +840,79 @@ private bool ExecuteInstruction()
                         break;
                     }
 
-                // Array
-                case OpCode.ARRAYSIZE:
+                // Compound-type
+                case OpCode.PACK:
+                    {
+                        if (!TryPop(out int size)) return false;
+                        if (size < 0 || size > context.EvaluationStack.Count)
+                            return false;
+                        VMArray array = new VMArray(ReferenceCounter);
+                        for (int i = 0; i < size; i++)
+                        {
+                            if (!TryPop(out StackItem item)) return false;
+                            array.Add(item);
+                        }
+                        Push(array);
+                        break;
+                    }
+                case OpCode.UNPACK:
+                    {
+                        if (!TryPop(out VMArray array)) return false;
+                        for (int i = array.Count - 1; i >= 0; i--)
+                            Push(array[i]);
+                        Push(array.Count);
+                        break;
+                    }
+                case OpCode.NEWARRAY0:
+                    {
+                        Push(new VMArray(ReferenceCounter));
+                        break;
+                    }
+                case OpCode.NEWARRAY:
+                case OpCode.NEWARRAY_T:
+                    {
+                        if (!TryPop(out int n)) return false;
+                        if (n < 0 || n > MaxStackSize) return false;
+                        StackItem item;
+                        if (instruction.OpCode == OpCode.NEWARRAY_T)
+                        {
+                            StackItemType type = (StackItemType)instruction.TokenU8;
+                            if (!Enum.IsDefined(typeof(StackItemType), type)) return false;
+                            item = instruction.TokenU8 switch
+                            {
+                                (byte)StackItemType.Integer => Integer.Zero,
+                                (byte)StackItemType.ByteArray => ByteArray.Empty,
+                                _ => StackItem.Null
+                            };
+                        }
+                        else
+                        {
+                            item = StackItem.Null;
+                        }
+                        Push(new VMArray(ReferenceCounter, Enumerable.Repeat(item, n)));
+                        break;
+                    }
+                case OpCode.NEWSTRUCT0:
+                    {
+                        Push(new Struct(ReferenceCounter));
+                        break;
+                    }
+                case OpCode.NEWSTRUCT:
```

Comment:
It's needed `NEWSTRUCT_T`?

Reply:
The types of fields in struct are usually different.